shell/plugins/directives.js

```diff
@@ -39,3 +29,28 @@ const getElement = (vnode) => {
     return componentInstance.$refs.input.$refs.value;
   }
 };
+
+function inserted(_el, _binding, vnode) {
+  const element = getElement(vnode);
+
+  if (element) {
+    element.focus();
+  }
+}
+
+const focus = {
+  install: (Vue) => {
+    if (Vue.directive('focus')) {
+      // eslint-disable-next-line no-console
+      console.debug('Skipping focus install. Directive already exists.');
+    } else {
+      Vue.directive('focus', { inserted });
+    }
+  }
+};
+
+export default focus;
+/* eslint-disable-next-line no-console */
+console.warn('The implicit addition of focus has been deprecated in Rancher Shell and will be removed in a future version. Make sure to invoke `Vue.use(focus)` to maintain compatibility.');
+
+Vue.use(focus);
```

Comment:
It might not be necessary to do this everywhere, given https://github.com/rancher/dashboard/pull/10923/files#diff-4964064b3ea3d66459e9eb386e2612aa8aea93c2d858eca54128348b498dcabb